crates/cheatcodes/src/evm/fork.rs

```diff
@@ -278,14 +278,18 @@ fn create_select_fork(ccx: &mut CheatsCtxt, url_or_alias: &str, block: Option<u6
     check_broadcast(ccx.state)?;
 
     let fork = create_fork_request(ccx, url_or_alias, block)?;
-    let id = ccx.ecx.db.create_select_fork(fork, &mut ccx.ecx.env, &mut ccx.ecx.journaled_state)?;
+    let id = ccx
+        .ecx
+        .db
+        .create_select_fork(fork, &mut ccx.ecx.env, &mut ccx.ecx.journaled_state)
+        .map_err(|err| fmt_err!("{err:?}"))?;
```

Comment:
If these are for eyre errors then i would revert these changes and update the implementation of `From<eyre::Error>` https://github.com/foundry-rs/foundry/blob/8bdcbfa4d65408b75c4038bd5ee67ce7f6dbd3bb/crates/cheatcodes/src/error.rs#L296

Reply:
Makes sense, this would be also fixing same issue for non vm errors like here https://github.com/foundry-rs/foundry/issues/8908#issuecomment-2422931839 if I get this right?

Reply:
yes

Reply:
@DaniPopes pls check https://github.com/foundry-rs/foundry/pull/9141/commits/c0ea8503db90f1bb9554f9ab9336422c54191d4b